crates/core_simd/src/masks/bitmask.rs

```diff
@@ -115,22 +115,34 @@ where
         unsafe { Self(intrinsics::simd_bitmask(value), PhantomData) }
     }
 
-    #[cfg(feature = "generic_const_exprs")]
+    // Safety: N must be the exact number of bytes required to hold the bitmask for this mask
     #[inline]
     #[must_use = "method returns a new array and does not mutate the original value"]
-    pub fn to_bitmask(self) -> [u8; LaneCount::<LANES>::BITMASK_LEN] {
+    pub unsafe fn to_bitmask_array<const N: usize>(self) -> [u8; N] {
```

Comment:
if you assert, you get a safe function afaict.
```suggestion
    pub fn to_bitmask_array<const N: usize>(self) -> [u8; N] {
        assert_eq!(N, LaneCount::<LANES>::BITMASK_LEN, "invalid bitmask array size");
```

Reply:
having transmute_copy of a wrong-sized object in a code-path that's never executed is at least just as safe as calling `unreachable_unchecked` -- perfectly safe imho.

```rust
pub fn this_is_safe() {
    assert!(1 == 2);
    unsafe { unreachable_unchecked() }
}
```

Reply:
`to_bitmask_array` feels like a more confusing name than what we had before (i.e. giving a bitmask, but as bytes) since it may imply, since many masks are 1 byte per lane, somehow lane masks, but as a set of bytes...? I think the "bitmask" terminology doesn't adequately distinguish it from the "width = lane" mask, since those are also used in the same ways that "bitmasks" are used elsewhere in programming.

If we are going to have such a verbose method, maybe `to_bitmask_as_bytes`.

Reply:
I'm okay with that name too, or any other suggestions.

Reply:
I'm not going to resolve this whole thread because of the naming issue, but the original concern about the function being safe has been addressed.